src/ResourceManager/SiteRecovery/Commands.SiteRecovery/Vault/SetAzureSiteRecoveryVaultSettings.cs

```diff
@@ -31,7 +31,7 @@ public class SetAzureSiteRecoveryVaultSettings : SiteRecoveryCmdletBase
         /// <summary>
         /// Gets or sets vault Object.
         /// </summary>
-        [Parameter(ParameterSetName = ASRParameterSets.Default, Mandatory = true)]
+        [Parameter(ParameterSetName = ASRParameterSets.Default, Mandatory = true, ValueFromPipeline = true)]
```

Comment:
This is very, very bad practice.  Everything on the pipeline is convertible to object